micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/ExecutorServiceMetrics.java

```diff
@@ -439,6 +448,39 @@ private void monitor(MeterRegistry registry, ForkJoinPool fj) {
         registeredMeterIds.addAll(meters.stream().map(Meter::getId).collect(toSet()));
     }
 
+    private void monitorThreadPerTaskExecutor(MeterRegistry registry, ExecutorService executorService) {
+        List<Meter> meters = asList(Gauge
+            .builder(metricPrefix + "executor.active", executorService,
+                    ExecutorServiceMetrics::getThreadCountFromThreadPerTaskExecutor)
+            .tags(tags)
+            .description("The approximate number of threads that are actively executing tasks")
+            .baseUnit(BaseUnits.THREADS)
+            .register(registry));
+        registeredMeterIds.addAll(meters.stream().map(Meter::getId).collect(toSet()));
+    }
+
+    private static long getThreadCountFromThreadPerTaskExecutor(ExecutorService executorService) {
+        try {
+            return (long) METHOD_THREAD_COUNT_FROM_THREAD_PER_TASK_EXECUTOR.invoke(executorService);
+        }
+        catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Nullable
+    private static Method getMethodForThreadCountFromThreadPerTaskExecutor() {
```

Comment:
Wouldn't it be better to generate a `java.lang.invoke.MethodHandle` once (maybe static) and use it (the same one, cached) in every call? Maybe it would be better for performance reasons

Reply:
@juliojgd Thanks for the feedback!

I'm not sure how easy implementing this with the `MethodHandle` would be, but it's not a hot spot, so I think it's okay as is.

Reply:
IMHO, the gauge will call the method every N seconds, right? And the method in turn will make the reflective call each time, so maybe it could be a hotspot (due to the number of calls made during the JVM instance life).

Reply:
@juliojgd I'm not familiar with the `MethodHandle`, so I'm not sure how effective it is here, but I attempted to switch to it in e5d159b.